x-pack/plugins/observability_ai_assistant/public/hooks/use_timeline.ts

```diff
@@ -89,6 +92,13 @@ export function useTimeline({
 
   const [pendingMessage, setPendingMessage] = useState<PendingMessage>();
 
+  const prevConversationId = usePrevious(conversationId);
+  useEffect(() => {
+    if (prevConversationId !== conversationId && pendingMessage?.error) {
+      setPendingMessage(undefined);
+    }
+  }, [conversationId, pendingMessage?.error, prevConversationId]);
```

Comment:
I don't think it only needs to be cleared when there is an error?

Reply:
Could you elaborate? What other scenarios do you think this should happen? Not seeing any obvious errors atm

Reply:
if you start a new conversation, and wait until it's stored, add a new message, and immediately click on "new chat" (while the assistant response is being loaded), the pending message is added there (in the new conversation). You can also use an existing conversation, do the same and then click on another conversation.